### PR TITLE
Fix protocol seq in AuthSwitchRequest 

### LINKED ISSUE #1878: Need help: stateless test fails on local mac
### Env:

With the latest master: b2f51d2446eee8cd086dbd1bca7ce89b7517ed25
`make stateless-test`

```
mysql --version
mysql  Ver 8.0.25 for macos10.15 on x86_64 (Homebrew)
```

### Failures:

It seems like every test fails. But unit tests all passed.
I am not sure if it is caused by some of my local lib/bin, e.g. mysql client, or by some incompatible dependency.

```
Starting databend-test

Running 45 stateless tests.

00_0000_dummy_select_1:                                                 [ FAIL ] - return code 1
, result:

ERROR 2013 (HY000): Lost connection to MySQL server during query
...
```

No logs output to `/_logs/`.

The query backtrace is as following( [nohup.out.zip](https://github.com/datafuselabs/databend/files/7189422/nohup.out.zip) ):
```
[2021-09-18T07:05:48Z ERROR databend_query::servers::mysql::mysql_session] Unexpected error occurred during query execution: Code: 1002, displayText = peer terminated connection.

       0: backtrace::backtrace::libunwind::trace
                 at /Users/drdrxp/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.61/src/backtrace/libunwind.rs:90:5
          backtrace::backtrace::trace_unsynchronized
                 at /Users/drdrxp/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.61/src/backtrace/mod.rs:66:5
       1: backtrace::backtrace::trace
                 at /Users/drdrxp/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.61/src/backtrace/mod.rs:53:14
       2: backtrace::capture::Backtrace::create
                 at /Users/drdrxp/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.61/src/capture.rs:176:9
       3: backtrace::capture::Backtrace::new
                 at /Users/drdrxp/.cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.61/src/capture.rs:140:22
       4: common_exception::exception::ErrorCode::from_std_error
                 at common/exception/src/exception.rs:408:65
       5: <common_exception::exception::ErrorCode as core::convert::From<std::io::error::Error>>::from
                 at common/exception/src/exception.rs:374:9
       6: <core::result::Result<T,F> as core::ops::try_trait::FromResidual<core::result::Result<core::convert::Infallible,E>>>::from_residual
                 at /rustc/b69fe57261086e70aea9d5b58819a1794bf7c121/library/core/src/result.rs:1915:27
       7: msql_srv::MysqlIntermediary<B,R,W>::init
                 at /Users/drdrxp/.cargo/git/checkouts/msql-srv-62b5153e0120bb3a/60e369b/src/lib.rs:420:54
       8: msql_srv::MysqlIntermediary<B,R,W>::run_on
                 at /Users/drdrxp/.cargo/git/checkouts/msql-srv-62b5153e0120bb3a/60e369b/src/lib.rs:307:9
       9: msql_srv::MysqlIntermediary<B,std::net::tcp::TcpStream,std::net::tcp::TcpStream>::run_on_tcp
                 at /Users/drdrxp/.cargo/git/checkouts/msql-srv-62b5153e0120bb3a/60e369b/src/lib.rs:274:9
      10: databend_query::servers::mysql::mysql_session::MySQLConnection::session_executor
                 at query/src/servers/mysql/mysql_session.rs:42:29
      11: databend_query::servers::mysql::mysql_session::MySQLConnection::run_on_stream::{{closure}}
                 at query/src/servers/mysql/mysql_session.rs:34:13
...
```

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Fix protocol seq in AuthSwitchRequest 

```
❯ mysql --version
mysql  Ver 8.0.19 for osx10.14 on x86_64 (Homebrew)


❯ mysql --host 192.168.101.40 --user default --port 3307
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 8
Server version: DatabendQuery v-0.1.0-13d03b2-simd(1.57.0-nightly-2021-09-20T03:42:02.090437433+00:00) 0

Copyright (c) 2000, 2020, Oracle and/or its affiliates. All rights reserved.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> select 1;
+------+
| 1    |
+------+
|    1 |
+------+
1 row in set (0.01 sec)
Read 0 rows, 0 B in 0.003 sec., 0 rows/sec., 0 B/sec.

mysql>
```

## Changelog

- Bug Fix

## Related Issues

https://github.com/datafuse-extras/msql-srv/pull/9

Fixes #1878 

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: query/Cargo.toml [hunk 1]
```diff
@@ -42,7 +42,7 @@ common-metatypes = { path = "../common/metatypes" }
 common-clickhouse-srv = { path = "../common/clickhouse-srv" }
 
 # Github dependencies
-msql-srv = { git = "https://github.com/datafuse-extras/msql-srv", rev = "60e369b" }
+msql-srv = { git = "https://github.com/datafuse-extras/msql-srv", rev = "9c706a3" }
```
Comment (BohuTANG) [inline query/Cargo.toml:45]: Why not merge https://github.com/datafuse-extras/msql-srv/pull/9?

Comment (BohuTANG) [reply query/Cargo.toml:45]: I merged

**Comment by drmingdrmer** [comment]
@sundy-li
Once again, you saved my ass! Thanks bro!

**Pull request merged**